### ISSUE TITLE
Plugin fails to build on Android in Phonegap Build

### ISSUE THREAD

**Issue by OverlappingElvis** [jbavari/cordova-plugin-video-editor#61]
I'm having some issues with this plugin on Android using the Phonegap Build service. iOS and Windows seem to build fine, but throws the following on Android:

```
Fetching plugin "cordova-plugin-video-editor" via npm
Installing "cordova-plugin-video-editor@1.0.5" for android
--------------------------------------------------------------------------------
COMPILE OUTPUT
--------------------------------------------------------------------------------
Executing: ant debug -f /project/build.xml
Buildfile: /project/build.xml
Buildfile: /project/build.xml

-set-mode-check:

-set-debug-files:

-check-env:
 [checkenv] Android SDK Tools Revision 24.4.1

-setup:
     [echo] Project Name: .ePals

-check-env:
 [checkenv] Android SDK Tools Revision 24.4.1

-setup:
     [echo] Project Name: .ePals
  [gettype] Project Type: Application

-set-debug-mode:

-debug-obfuscation-check:

-pre-build:

-build-setup:
  [gettype] Project Type: Application

-pre-clean:

clean:
[getlibpath] Library dependencies:
[getlibpath] 
[getlibpath] ------------------
[getlibpath] Ordered libraries:
[getbuildtools] Using latest Build Tools: 23.0.1
     [echo] Resolving Build Target for .ePals...

nodeps:

-check-env:
 [checkenv] Android SDK Tools Revision 24.4.1

-setup:
     [echo] Project Name: .ePals
  [gettype] Project Type: Android Library

-pre-clean:

clean:
[gettarget] Project Target:   Android 5.1.1
[gettarget] API level:        22
     [echo] ----------
     [echo] Creating output directories if needed...
    [mkdir] Created dir: /project/bin
    [mkdir] Created dir: /project/bin/res
    [mkdir] Created dir: /project/bin/rsObj
    [mkdir] Created dir: /project/bin/rsLibs
    [mkdir] Created dir: /project/gen
    [mkdir] Created dir: /project/bin/classes
    [mkdir] Created dir: /project/bin/dexedLibs
     [echo] ----------
     [echo] Resolving Dependencies for .ePals...
[dependency] Library dependencies:
[dependency] 
[dependency] ------------------
[dependency] Ordered libraries:
[dependency] 
[dependency] ------------------
     [echo] ----------
     [echo] Building Libraries with 'debug'...

nodeps:

-check-env:
 [checkenv] Android SDK Tools Revision 24.4.1

-setup:
     [echo] Project Name: .ePals
  [gettype] Project Type: Android Library

-pre-clean:

clean:

BUILD SUCCESSFUL
Total time: 1 second

nodeps:

-set-mode-check:

-set-debug-files:

-check-env:
 [checkenv] Android SDK Tools Revision 24.4.1

-setup:
     [echo] Project Name: .ePals
  [gettype] Project Type: Android Library

-set-debug-mode:

-debug-obfuscation-check:

-pre-build:

-build-setup:
[getbuildtools] Using latest Build Tools: 23.0.1
     [echo] Resolving Build Target for .ePals...
[gettarget] Project Target:   Android 5.1.1
[gettarget] API level:        22
     [echo] ----------
     [echo] Creating output directories if needed...
    [mkdir] Created dir: /project/cordova-plugin-video-editor/epals2-android-ffmpeg-java/libs
    [mkdir] Created dir: /project/cordova-plugin-video-editor/epals2-android-ffmpeg-java/bin
    [mkdir] Created dir: /project/cordova-plugin-video-editor/epals2-android-ffmpeg-java/bin/res
    [mkdir] Created dir: /project/cordova-plugin-video-editor/epals2-android-ffmpeg-java/bin/rsObj
    [mkdir] Created dir: /project/cordova-plugin-video-editor/epals2-android-ffmpeg-java/bin/rsLibs
    [mkdir] Created dir: /project/cordova-plugin-video-editor/epals2-android-ffmpeg-java/gen
    [mkdir] Created dir: /project/cordova-plugin-video-editor/epals2-android-ffmpeg-java/bin/classes
    [mkdir] Created dir: /project/cordova-plugin-video-editor/epals2-android-ffmpeg-java/bin/dexedLibs
     [echo] ----------
     [echo] Resolving Dependencies for .ePals...
[dependency] Library dependencies:
[dependency] No Libraries
[dependency] 
[dependency] ------------------

-code-gen:
[mergemanifest] Merging AndroidManifest files into one.
[mergemanifest] Manifest merger disabled. Using project manifest only.
     [echo] Handling aidl files...
     [aidl] No AIDL files to compile.
     [echo] ----------
     [echo] Handling RenderScript files...
     [echo] ----------
     [echo] Handling Resources...
     [aapt] Generating resource IDs...
     [echo] ----------
     [echo] Handling BuildConfig class...
[buildconfig] Generating BuildConfig class.

-pre-compile:

-compile:
    [javac] Compiling 22 source files to /project/cordova-plugin-video-editor/epals2-android-ffmpeg-java/bin/classes
     [echo] Creating library output jar file...
      [jar] Building jar: /project/cordova-plugin-video-editor/epals2-android-ffmpeg-java/bin/classes.jar

-post-compile:

BUILD FAILED
/android-sdk/tools/ant/build.xml:597: The following error occurred while executing this line:
/project/cordova-plugin-video-editor/epals2-android-ffmpeg-java/custom_rules.xml:4: /project/cordova-plugin-video-editor/epals2-android-ffmpeg-java/ant-build does not exist.

Total time: 4 seconds

/project/cordova/node_modules/q/q.js:126
                    throw e;
                    ^
Error code 1 for command: ant with args: debug,-f,/project/build.xml
```

**Comment by rossmartin** [comment]
Can you provide your apps config.xml file?  Thanks.

**Comment by OverlappingElvis** [comment]
Sure:

```
<?xml version="1.0" encoding="UTF-8" ?>
<widget xmlns="http://www.w3.org/ns/widgets" xmlns:gap="http://phonegap.com/ns/1.0" xmlns:android="http://schemas.android.com/apk/res/android" id="com.ieq.epals2" version="__version__">
	<name>ePals</name>
	<description>Cordova container.</description>
	<access origin="*" />
	<gap:plugin name="cordova-plugin-whitelist" source="npm" />
	<gap:plugin name="cordova-plugin-network-information" source="npm" />
	<gap:plugin name="cordova-plugin-media-capture-vibi" source="npm" />
	<gap:plugin name="cordova-plugin-camera" version="1.2.0" source="npm" />
	<gap:plugin name="cordova-plugin-file" source="npm" />
	<gap:plugin name="cordova-plugin-file-transfer" version="1.5.0" source="npm" />
	<gap:plugin name="cordova-plugin-x-toast" source="npm" />
	<gap:plugin name="cordova-plugin-device" source="npm" />
	<gap:plugin name="cordova-plugin-insomnia" source="npm" />
	<gap:plugin name="cordova-plugin-video-editor" source="npm" />
	<icon src="icon.png" />
	<!-- iOS icon sizes -->
	<icon src="icon-40.png" gap:platform="ios" width="40" height="40" />
	<icon src="icon-40@2x.png" gap:platform="ios" width="80" height="80" />
	<icon src="icon-57.png" gap:platform="ios" width="57" height="57" />
	<icon src="icon-57@2x.png" gap:platform="ios" width="114" height="114" />
	<icon src="icon-72.png" gap:platform="ios" width="72" height="72" />
	<icon src="icon-72@2x.png" gap:platform="ios" width="144" height="144" />
	<icon src="icon-76.png" gap:platform="ios" width="76" height="76" />
	<icon src="icon-76@2x.png" gap:platform="ios" width="152" height="152" />

	<!-- iPhone and iPod touch -->
	<gap:splash src="Default.png" gap:platform="ios" width="320" height="480" />
	<gap:splash src="Default@2x.png" gap:platform="ios" width="640" height="960" />
	<icon src="icon-57.png" gap:platform="ios" width="57" height="57" />
	<icon src="icon-72.png" gap:platform="ios" width="72" height="72" />

	<!-- iPhone 5 / iPod Touch (5th Generation) -->
	<gap:splash src="Default-568h@2x.png" gap:platform="ios" width="640" height="1136" />

	<!-- iPhone 6 -->
	<gap:splash src="Default-667h@2x.png" gap:platform="ios" width="750" height="1334" />
	<gap:splash src="Default-Portrait-736h@3x.png" gap:platform="ios" width="1242" height="2208" />
	<gap:splash src="Default-Landscape-736h@3x.png" gap:platform="ios" width="2208" height="1242" />

	<!-- iPad -->
	<gap:splash src="Default-Portrait.png" gap:platform="ios" width="768" height="1024" />
	<gap:splash src="Default-Landscape.png" gap:platform="ios" width="1024" height="768" />

	<!-- Retina iPad -->
	<gap:splash src="Default-Portrait@2x.png" gap:platform="ios" width="1536" height="2048" />
	<gap:splash src="Default-Landscape@2x.png" gap:platform="ios" width="2048" height="1536" />

	<!-- Android -->
	<gap:splash src="ldpi.png" gap:platform="android" gap:qualifier="ldpi" />
	<gap:splash src="mdpi.png" gap:platform="android" gap:qualifier="mdpi" />
	<gap:splash src="hdpi.png" gap:platform="android" gap:qualifier="hdpi" />
	<gap:splash src="xhdpi.png" gap:platform="android" gap:qualifier="xhdpi" />

	<!-- Hide title bar -->
	<preference name="fullscreen" value="true" />
	<gap:config-file platform="android" parent="/manifest">
		<application android:theme="@android:style/Theme.NoTitleBar">
			<activity android:theme="@android:style/Theme.NoTitleBar.Fullscreen"></activity>
		</application>
	</gap:config-file>
	<preference name="StatusBarOverlaysWebView" value="false" />

	<preference name="iosPersistentFileLocation" value="Library" />

	<feature name="http://api.phonegap.com/1.0/file"/>
	<feature name="http://api.phonegap.com/1.0/device"/>
	<feature name="http://api.phonegap.com/1.0/network"/>

</widget>
```

This has only been happening since adding `cordova-plugin-video-editor`

**Comment by rossmartin** [comment]
Here is the contents of the `/project/cordova-plugin-video-editor/${project}-android-ffmpeg-java/custom_rules.xml` -

```
<?xml version="1.0" encoding="UTF-8"?>
<project>
    <target name="-post-compile">
        <copy todir="bin">
            <fileset dir="ant-build"/>
        </copy>
    </target>
</project>
```

For some reason phonegap build does not like this (I have never used phonegap build either).  I'd be curious what would happen if you forked the repo and either deleted this custom_rules.xml file or commented out the contents in it.

Could you fork the repo, do this, and have phonegap build point to your custom fork on github?

**Comment by OverlappingElvis** [comment]
Yes, I'll give it a shot!

**Comment by rossmartin** [comment]
The official [phonegap-facebook-plugin](https://github.com/Wizcorp/phonegap-facebook-plugin) uses this same `custom_rules.xml` (exact same contents too).

I remember at time I had to add this `custom_rules.xml` file to be able to build with the cordova CLI on android.  

What's interesting is that they don't require anything special to use with phonegap build (at least from the looks of it) - https://github.com/Wizcorp/phonegap-facebook-plugin/blob/master/platforms/pg-build/README.md

**Comment by OverlappingElvis** [comment]
Yeah, very strange — I'll update this issue as well if I find anything.

**Comment by OverlappingElvis** [comment]
@rossmartin Good thinking — removing `custom_rules.xml` seems to let the build finish, but I don't know what the ramifications are for removing the copy step and am testing that.

**Comment by OverlappingElvis** [comment]
@rossmartin this seems to be working as expected under Phonegap Build.

**Comment by rossmartin** [comment]
I'll make sure this still builds with the cordova CLI and make the change locally.

I remember I had to add this file to get it to build with the CLI but that was prior to the CLI using gradle, it was using ant then. It may not matter anymore.

**Comment by rossmartin** [comment]
Thanks for letting me know that it works in phonegap build now.  I should have time tonight to test this change with the cordova CLI.

**Comment by rossmartin** [comment]
Thanks, closed by #62.  I did confirm that you can you still build with the cordova CLI.
